.pre-commit-config.yaml

```diff
@@ -0,0 +1,22 @@
+exclude: ^docs/
+
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.5.0
+  hooks:
+  - id: check-yaml
+  - id: check-case-conflict
+  - id: check-json
+  - id: check-merge-conflict
+  - id: check-symlinks
+  - id: check-toml
+  - id: end-of-file-fixer
```

Comment:
when running pre-commit locally I saw that it was fixing files that did not have a newline at the end, but in the review saw that some of the files did not have a newline, please run `pre-commit run --all-files` once and commit the results